app/models/issue.rb

```diff
@@ -45,4 +45,16 @@ def lt_ln_present?(obj)
   def status_name
     STATUSES[status]
   end
+
+  def published?
+    (status == 'open') || (status == 'closed')
+  end
+
+  def can_read_by?(user)
+    if user.present?
+      user.admin? || user.moderator? || (self.user == user) || published?
+    else
+      false
+    end
+  end
```

Comment:
yes ... but it's also not useful, I mean additional check also will be needed, for example to check if user is persisted

overriding current_user is not something what devise recommends